smarts/core/remote_agent_buffer.py

```diff
@@ -93,6 +93,16 @@ def __init__(self, zoo_manager_addrs=None, buffer_size=3, max_workers=4):
             self._remote_agent_future() for _ in range(self._buffer_size)
         ]
 
+        # Catch abrupt terminate signals
+        signal.signal(signal.SIGTERM, self._stop_servers)
```

Comment:
Is there any need to unset the signal when the termination handler is called?

Reply:
Probably. Need investigation on multiple calls to terminate.